py3status/modules/hueshift.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+"""
+Shift color temperature on the screen.
+
+Configuration parameters:
+    button_down: mouse button to change color temperature (default 5)
+    button_reset: mouse button to reset color temperature (default 3)
+    button_up: mouse button to change color temperature (default 4)
+    command: specify blueshift, redshift, or sct to use, otherwise automatic
+        (default None)
+    delta: specify interval value to change color temperature (default 250)
+    format: display format for this module
+        *(default '\?if=disabled&color=bad \u263c!'
+        '|\?color=hue \u263c {temperature}K')*
+    initial: specify initial color temperature to use (default 6500)
+    maximum: specify maximum color temperature to use (default 25000)
+    minimum: specify minimum color temperature to use (default 1000)
+    thresholds: specify color thresholds to use
+        (default [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')])
+
+Control placeholder:
+    {disabled} a boolean based on pgrep processing data
+
+Format placeholder:
+    {temperature} color temperature, eg 6500
+    {command} command name, eg blueshift, redshift, sct
+
+Color options:
+    color_bad: hueshift paused due to running processes
+
+Color thresholds:
+    hue: print a color based on the value of color temperature
+
+Requires:
+    blueshift: an extensible and highly configurable alternative
+    redshift: program to adjust the color temperature of your screen
+    sct: set color temperature
+
+Suggestions:
+    campfire: 4500
+    dust storm on mars: 2000
+    coffee free all nighter: 8000
+
+Examples:
+```
+# good rules; show command name; don't colorize everything
+hueshift {
+    format = '[\?if=disabled&color=bad \u263c!|[\?color=hue&show \u263c '
+    format += '{command}] {temperature}K]'
+    minimum = 3000
+    maximum = 10000
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+# warm
+{'full_text': u'\u263c 3000K', 'color': #f3c},
+
+# neutral
+{'full_text': u'\u263c 6500K', 'color': #ff3}
+
+# cool
+{'full_text': u'\u263c 10000K', 'color': #3cf},
+
+disabled
+{'full_text': u'\u263c!', 'color': #f00}
+"""
+
+STRING_BAD_COMMAND = 'invalid command `%s`'
+STRING_NOT_INSTALLED = 'command `%s` not installed'
+STRING_NOT_AVAILABLE = 'no available command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_reset = 3
+    button_up = 4
+    command = None
+    delta = 250
+    format = u'\?if=disabled&color=bad \u263c!|\?color=hue \u263c {temperature}K'
+    initial = 6500
+    maximum = 25000
+    minimum = 1000
+    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
```

Comment:
these default thresholds feel very tame I'd prefer something a bit more noticeable and maybe feedback (I realise the screen color is fairly strong feedback in ways but our eyes adjust color perception massively.

maybe something more like `thresholds = [(4000, '#FF0000'), (6500, '#FFFFFF'), (10000, '#0000FF')]`

Reply:
The colors  `[(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]` comes from some website I researched to make `hueshift` available with other binaries. I think this color is nice, reasonable, and soft on eyes versus a French flag. I want to keep it as-is. Please. :-)

Reply:
you should use gradients :)

Reply:
After playing with `gradients`, I think we gain nothing for the users and would rather make it obvious... only 3 colors.

Also, I think my `pink` game is a bit too strong for others so please try the following thresholds and pick a choice that works nice for you.

For me, 3rd is pretty and fluffy. Maybe a tad too light pink.
I feel 2nd is reasonable / acceptable for most users.
1st is original. The pink is strong with this one.
```
    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
    thresholds = [(6499, '#f6c'), (6500, '#ff6'), (6501, '#6cf')]
    thresholds = [(6499, '#f9c'), (6500, '#ff9'), (6501, '#9cf')]
```
I can't believe I'm discussing pink with @tobes again.

Reply:
Nice thing about storage is that you can refresh and still in same place... so this is much easier to test colors too.